### PR TITLE
fix: ping a node before attempting to remove or replace

### PR DESCRIPTION
**Pull request by AlCalzone**
Based on https://github.com/zwave-js/node-zwave-js/discussions/1447 this is how we should handle the node removal/replacement process.

closes: #1444 

@robertsLando you can remove the isFailed check if you do that before calling these methods.

### CODE REVIEW

File: packages/zwave-js/src/lib/controller/Controller.ts [hunk 1]
```diff
@@ -1885,6 +1885,21 @@ ${associatedNodes.join(", ")}`,
...
+				ZWaveErrorCodes.Controller_NodeNotFound,
+			);
+		}
+
+		if (await node.ping()) {
+			throw new ZWaveError(
+				`The node removal process could not be started because the node responded to a ping.`,
+				ZWaveErrorCodes.ReplaceFailedNode_Failed,
+			);
+		}
```
Comment (robertsLando) [inline packages/zwave-js/src/lib/controller/Controller.ts]: This is repeated later, couldn't this be a new private function like `pingNode` taht returns a node?

Comment (AlCalzone) [reply packages/zwave-js/src/lib/controller/Controller.ts]: I don't follow? The second part is in another method.
I could extract a helper getNodeOrThrow but that should be separate from the ping

File: packages/zwave-js/src/lib/controller/Controller.ts [hunk 2]
```diff
@@ -1970,6 +1985,21 @@ ${associatedNodes.join(", ")}`,
...
+				ZWaveErrorCodes.Controller_NodeNotFound,
+			);
+		}
+
+		if (await node.ping()) {
+			throw new ZWaveError(
+				`The node replace process could not be started because the node responded to a ping.`,
+				ZWaveErrorCodes.ReplaceFailedNode_Failed,
+			);
+		}
```
Comment (robertsLando) [inline packages/zwave-js/src/lib/controller/Controller.ts]: Here you use the same code you used previously, could be a function so

Comment (AlCalzone) [reply packages/zwave-js/src/lib/controller/Controller.ts]: Different error message. I don't see the benefit in having a method like getNodeOrThrowAndThrowIfPingSuceeded.
The first part, okay.

**Comment by AlCalzone** [comment]
Refactored based on #1457 to address review comments

**Review by robertsLando — DISMISSED** [review]
Yep much cleaner now! LGTM :)

**Comment by AlCalzone** [comment]
@zwave-js-bot rebase

**Comment by AlCalzone** [comment]
try again, @zwave-js-bot rebase

**Comment by AlCalzone** [comment]
try again, @zwave-js-bot rebase

**Pull request merged**